src/extensionConsole/pages/mods/ModsPage.test.tsx

```diff
@@ -0,0 +1,290 @@
+const useOptionalModDefinitionMock = jest.mocked(useOptionalModDefinition);
+
+jest.mock("@/extensionConsole/pages/deployments/useAutoDeploy");
+const useAutoDeployMock = jest.mocked(useAutoDeploy);
+
+jest.mock("@/extensionConsole/pages/mods/onboardingView/useOnboarding");
+const useOnboardingMock = jest.mocked(useOnboarding);
+
+describe("ModsPage", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    useAllModDefinitionsMock.mockReturnValue(valueToAsyncCacheState([]));
+    useOptionalModDefinitionMock.mockReturnValue(
+      valueToAsyncCacheState(undefined),
+    );
+    useAutoDeployMock.mockReturnValue({ isAutoDeploying: false });
+    appApiMock.reset();
+    appApiMock.onGet(API_PATHS.MARKETPLACE_LISTINGS).reply(200, []);
+    appApiMock.onGet(API_PATHS.FEATURE_FLAGS).reply(200, {
+      flags: [],
+    });
+    appApiMock.onGet(API_PATHS.BRICKS).reply(200, []);
+  });
+
+  it("renders loading state when mods are loading", async () => {
+    useAllModDefinitionsMock.mockReturnValue(loadingAsyncCacheStateFactory());
+    useOptionalModDefinitionMock.mockReturnValue(
+      loadingAsyncCacheStateFactory(),
+    );
+
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(true));
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId("loader")).toBeInTheDocument();
+    });
+  });
+
+  it("renders loading state when auto deploying", async () => {
+    useAutoDeployMock.mockReturnValue({ isAutoDeploying: true });
+
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId("loader")).toBeInTheDocument();
+    });
+  });
+
+  it("renders error state when mods fail to load", async () => {
+    useAllModDefinitionsMock.mockReturnValue(
+      errorToAsyncCacheState(new Error("Failed to load mods")),
+    );
+
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(false));
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText("An error occurred")).toBeInTheDocument();
+    });
+    expect(screen.getByText("Failed to load mods")).toBeInTheDocument();
+  });
+
+  it("renders error state when marketplace listings fail to load", async () => {
+    appApiMock.onGet(API_PATHS.MARKETPLACE_LISTINGS).reply(500);
+
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(false));
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText("An error occurred")).toBeInTheDocument();
+    });
```

Comment:
NIT: This is the "correct" way to write this:
```suggestion
    await expect(screen.findByText("An error occurred")).resolves.toBeInTheDocument();
```